pkg/stanza/operator/input/promtail/promtail_test.go

```diff
@@ -0,0 +1,173 @@
+package promtail
+
+import (
+	"testing"
+	"time"
+
+	"github.com/grafana/loki/clients/pkg/promtail/api"
+	"github.com/grafana/loki/clients/pkg/promtail/positions"
+	"github.com/grafana/loki/clients/pkg/promtail/scrapeconfig"
+	"github.com/grafana/loki/clients/pkg/promtail/targets/file"
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/stretchr/testify/require"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/entry"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/testutil"
+)
+
```

Comment:
You may want to use the `operatortest` package to validate unmarshaling of the config. The nice thing about this package is that it hooks into the collector's actual method of unmarshaling, so you are validating against all of its custom decode hooks. I found this very useful when adding mapstructure support to other operators.

[Here's an example](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/71dd19d2e59cd1f8aa9844461089d5c17efaa0ca/pkg/stanza/operator/parser/syslog/config_test.go#L28).